src/pyats/contrib/creators/topology.py

```diff
@@ -0,0 +1,775 @@
+import os
+import re
+import sys
+import time
+import pprint
+import logging
+import argparse
+import ipaddress
+import subprocess
+from itertools import product
+from collections import OrderedDict
+from concurrent.futures import ThreadPoolExecutor
+from yaml import (ScalarEvent, MappingEndEvent, MappingStartEvent, AliasEvent,
+                  SequenceEndEvent, SequenceStartEvent, DocumentEndEvent,
+                  DocumentStartEvent, StreamEndEvent, StreamStartEvent, emit,
+                  resolver, Dumper, dump, YAMLError, safe_load)
+
+from genie.conf import Genie
+from genie.testbed import load
+from pyats.async_ import pcall
+from .creator import TestbedCreator
+from genie.conf.base import Testbed, Device, Interface, Link
+from genie.metaparser.util.exceptions import SchemaEmptyParserError
+
+log = logging.getLogger(__name__)
+    
+supported_os = ['nxos', 'iosxr', 'iosxe', 'ios']
+
+
+
+class Topology(TestbedCreator):
+    
+    """ Topology class (TestbedCreator)
+
+        Takes a yaml file given by argument testbed_name and attempts to connect
+        to each device in the testbed and discover the devices connection using
+        cdp and lldp and writes it to a new  yaml file
+        Args:
+            testbed-name: Mandatory argument of initial testbed file
+            config: if enabled the script will enable cdp and lldp on devices
+                    and disable it afterwards
+            all-interfaces: if enabled, script will add all interfaces to 
+                    topology section of yaml file instead of just interfaces
+                    with active connections
+            exclude_network: list networks that won't be recorded by creator
+                    if found as part of a connection
+            exclude_interfaces:list interfaces that won't be recorded by creator
+                    if found as part of a connection
+            topo-only: if true script will only find connections between devices
+                    in the testbed and will not discover new devices
+            alias: takes argument in format device:alias,device2:alias2 and
+                    and indicates which alias should be used to connect to the 
+                    device first
+    """
+        
+    def _init_arguments(self):
+        """ Specifies the arguments for the creator.
+        Returns:
+            dict: Arguments for the creator.
+        """
+        # create 3 sets to track what devices have cdp and lldp configured
+        # and what devices have be visited by the script
+        self.cdp_configured = set()
+        self.lldp_configured = set()
+        self.visited_devices = set()
+        self.ad = {}

```

Comment:
what this is for? better to add comment or use more understandable variable name

Reply:
ad is supposed to hold the dictionary format of the alias argument, there's a small part of the _generate method that fills it up, I've renamed it alias_dict